feat: throw exception when trying to access message content without MESSAGE_CONTENT intent

Azn9 opened the pull request:
**Description:** Throw an UnsupportedOperationException when calling `Message#getContent` without the `MESSAGE_CONTENT` intent enabled.

**Justification:** Currently, this just returns an empty string, which users don't understand and might complain about it being a bug. This exception provides a detailed message, redirecting the user to this project's README, where a section has been added to address this issue.

darichey reviewed (approved):
I like it 👍

NovaFox161 commented:
While I support the sentiment of this, there's some edge cases this would break. Bots will get content for messages they are mentioned in, as well as any messages that are tied to an interaction, and the content of any messages authored by the bot. Additionally bots are always able to see message content in DMs.


If we could reliably handle that, then this change would be good to add.

Azn9 commented:
> While I support the sentiment of this, there's some edge cases this would break. Bots will get content for messages they are mentioned in, as well as any messages that are tied to an interaction, and the content of any messages authored by the bot. Additionally bots are always able to see message content in DMs.
> 
> If we could reliably handle that, then this change would be good to add.

Oh I didn't think about that oops. I'll look to add it!

Azn9 commented:
I added more checks for these case, and added the check to `Message#getEmbeds`, `Message#getAttachments` and `Message#getComponents` as these fields are also concerned by the needed intent ;)

NovaFox161 commented:
Thank you!

Doc94 commented:
Nice, maybe can be mentioned too in docs when merge? https://docs.discord4j.com/
